### PR TITLE
Setup tests with react testing library

### PR DESCRIPTION
**Pull request by karelhala**
Let's start using react testing library and properly setup tests to use typescript. This way the codebase will force typescript on every part of the code.

### CODE REVIEW

**Review by vidyanambiar — COMMENTED** [review]
Looks good!

**Comment by florkbr** [comment]
Not sure why the test job failed with sigkill:
```
PASS src/sdk/wsConfigs.test.ts (5.924 s)
  ✓ should properly get WS subprotocols (6 ms)
error Command failed with signal "SIGKILL".
```

File: frontend/jest.config.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import type { Config } from '@jest/types';
```
Comment (florkbr) [inline frontend/jest.config.ts:1]: As a follow up to your PR here - I really would like to de-dupe the jest config from hac-dev, hac-core, hac-bs, etc. Maybe we can do that through the SDK. At least that way our configs will be consistent and not vary across repos. 

https://github.com/openshift/dynamic-plugin-sdk/blob/main/jest.config.ts
https://github.com/openshift/hac-dev/blob/main/jest.config.js
https://github.com/RedHatInsights/frontend-components/blob/master/jest.config.js
https://github.com/openshift/console/blob/master/frontend/packages/eslint-plugin-console/lib/config/jest.js

**Comment by florkbr** [comment]
Still failing. @karelhala have we confirmed the return status of the jest command is correct? I also wonder if we need to fix the worker thread issue that we had in the SDK repo: https://github.com/openshift/dynamic-plugin-sdk/pull/106/files#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R13

```
PASS src/sdk/wsConfigs.test.ts (11.853 s)
  ✓ should properly get WS subprotocols (116 ms)
error Command failed with signal "SIGKILL".
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
{"component":"entrypoint","error":"wrapped process failed: exit status 1","file":"k8s.io/test-infra/prow/entrypoint/run.go:80","func":"k8s.io/test-infra/prow/entrypoint.Options.Run","level":"error","msg":"Error executing test process","severity":"error","time":"2022-06-01T17:37:57Z"} 
INFO[2022-06-01T17:38:01Z] Ran for 7m26s                                
ERRO[2022-06-01T17:38:01Z] Some steps failed:                           
ERRO[2022-06-01T17:38:01Z] 
  * could not run steps: step test failed: test "test" failed: the pod ci-op-2nttsdnz/test failed after 5m47s (failed containers: test): ContainerFailed one or more containers exited
```

**Comment by karelhala** [comment]
@florkbr looks like limiting the number of workers fixed the error. That's kinda strange, since you want to have as many workers available as CI gives you, but let's roll with it.

**Pull request merged**